### ISSUE TITLE
Error "Specified group generation id is not valid" during consumption

### ISSUE THREAD

**Issue by kiessan** [mateusjunges/laravel-kafka#295]
Hi there,

We quite often experience the error "Specified group generation id is not valid" when consuming messages for some of our topics:

```
{"throwable":{"file":"/app/vendor/mateusjunges/laravel-kafka/src/Commit/KafkaCommitter.php:20","code":22,"message":"Broker: Specified group generation id is not valid","class":"RdKafka\\Exception"}}
```

Unfortunately, this causes the consumer to break/fail. We fiddled around  with the heartbeat and timeout settings as suggested [here](https://github.com/tulios/kafkajs/issues/1009). Sadly, it doesn't solve the issue.

Question: Is it possible for the consumer to rejoin the group automatically? If not, any other solution than recreating the consumer?

Thx a lot for your advice!

**Comment by jesse-dijkstra** [comment]
I gues this could be the potential fix: https://github.com/tulios/kafkajs/pull/1474/commits/5ec2806d758425b9df17a7bcb348d37fdfd5be16

**Comment by rhu713** [comment]
We encountered the "Specified group generation id is not valid" as well while manually committing offsets via `Consumer.commitOffsets()`. It seems that any errors encountered during this, including the `generation id` error, are wrapped within a non-retriable error. We solved this by catching and unwrapping the unretriable error and re-throwing the underlying error, thus allowing https://github.com/tulios/kafkajs/commit/5ec2806d758425b9df17a7bcb348d37fdfd5be16 to handle the rejoining.
